bpf/lib/lb.h

```diff
@@ -1432,7 +1434,12 @@ lb4_select_backend_id_maglev(struct __ctx_buff *ctx __maybe_unused,
 			     const struct ipv4_ct_tuple *tuple,
 			     const struct lb4_service *svc)
 {
-	__be16 sport = lb4_svc_is_affinity(svc) ? 0 : tuple->sport;
+	/* CT tuple is swapped, hence the port swap. See also lb4_local()
+	 * call to ct_lazy_lookup4(..., CT_SERVICE, SCOPE_REVERSE, ...)
+	 * which then calls ipv4_ct_tuple_reverse().
+	 */
+	__be16 sport = lb4_svc_is_affinity(svc) ? 0 : tuple->dport;
+	__be16 dport = tuple->sport;
```

Comment:
heh yes, this is tricky. Passing around CT tuples needs quite a bit of context on how the caller set up the tuple.